devicetypes/plaidsystems/spruce-controller.src/spruce-controller.groovy

```diff
@@ -14,736 +10,553 @@
+			else sendCmds.push(zigbee.writeAttribute(ON_OFF_CLUSTER, OFF_WAIT_TIME_ATTRIBUTE, DataType.UINT16, runTime, [destEndpoint: endpoint]))
+		}
+	}
+	return sendCmds
 }
-def z13off() {
-	"st cmd 0x${device.deviceNetworkId} 14 6 0 {}"    
+
+//send switch time
+def writeType(endpoint, cycle) {
+	zigbee.writeAttribute(ON_OFF_CLUSTER, ON_TIME_ATTRIBUTE, DataType.UINT16, cycle, [destEndpoint: endpoint])
 }
-def z14on() {
-	return manual() + "st cmd 0x${device.deviceNetworkId} 15 6 1 {}"
+
+//send switch off time
+def writeTime(endpoint, runTime) {
+	zigbee.writeAttribute(ON_OFF_CLUSTER, OFF_WAIT_TIME_ATTRIBUTE, DataType.UINT16, runTime, [destEndpoint: endpoint])
 }
-def z14off() {
-    "st cmd 0x${device.deviceNetworkId} 15 6 0 {}"
+
+//set reporting and binding
+def configure() {
+	// Device-Watch allows 2 check-in misses from device, checks every 2 hours
+	sendEvent(name: "DeviceWatch-DeviceStatus", value: "online")
+	sendEvent(name: "healthStatus", value: "online")
+	sendEvent(name: "DeviceWatch-Enroll", value: 2* 60 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+
+	configureHealthCheck()
+
+	if (DEBUG) log.debug "Configuring Reporting and Bindings ${device.name} ${device.deviceNetworkId} ${device.hub.zigbeeId}"
+
+	//setup binding for 18 endpoints
+	def bindCmds = []
+	bindCmds += zigbee.addBinding(ON_OFF_CLUSTER, [destEndpoint: 1])
+	bindCmds += zigbee.addBinding(ALARMS_CLUSTER, [destEndpoint: 1])
+
+	for (endpoint in 1..18){
+		bindCmds += zigbee.addBinding(BINARY_INPUT_CLUSTER, [destEndpoint: endpoint])
+	}
+
+	//setup reporting for 18 endpoints
+	def reportingCmds = []
+	reportingCmds += zigbee.configureReporting(ON_OFF_CLUSTER, 0, DataType.BOOLEAN, 1, 0, 0x01, [destEndpoint: 1])
+	reportingCmds += zigbee.configureReporting(ALARMS_CLUSTER, 0, DataType.UINT16, 1, 0, 0x00, [destEndpoint: 1])
+
+	for (endpoint in 1..18){
+		reportingCmds += zigbee.configureReporting(BINARY_INPUT_CLUSTER, PRESENT_VALUE_IDENTIFIER, DataType.BOOLEAN, 1, 0, 0x01, [destEndpoint: endpoint])
+	}
+
+	return bindCmds + reportingCmds + setRainSensor()
 }
-def z15on() {
-	return manual() + "st cmd 0x${device.deviceNetworkId} 16 6 1 {}"  
+
+//PING is used by Device-Watch in attempt to reach the Device
+def ping() {
+	if (DEBUG) log.debug "device health ping"
+	return zigbee.onOffRefresh()
 }
-def z15off() {
-	"st cmd 0x${device.deviceNetworkId} 16 6 0 {}"    
+
+def refresh() {
+	if (DEBUG) log.debug "refresh"
+
+	def refreshCmds = []
+	for (endpoint in 1..17){
+		refreshCmds += zigbee.readAttribute(BINARY_INPUT_CLUSTER, PRESENT_VALUE_IDENTIFIER, [destEndpoint: endpoint])
+	}
+	refreshCmds += zigbee.readAttribute(BINARY_INPUT_CLUSTER, OUT_OF_SERVICE_IDENTIFIER, [destEndpoint: 18])
+
+	return refreshCmds
 }
-def z16on() {
-	return manual() + "st cmd 0x${device.deviceNetworkId} 17 6 1 {}"    
+
+def healthPoll() {
```

Comment:
Instead of this method, you should have `ping()` method implemented.

Generally speaking, you don't need any code related to keeping device online. 
1) Include `capability "Health Check"`, 
2) Define checkInterval with
`[name: "checkInterval", value: hcIntervalMinutes * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID]]`
3) Implement `ping()` method.

Reply:
ok, changed to ping and checkInterval is in configure